### PR TITLE
simplify external-provider logic

### PR DESCRIPTION
**Pull request by StanislavSava**

### CODE REVIEW

**Review by ioandanpopa — APPROVED** [review]

File: src/providers/accountProvider.ts [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
   accountProvider = provider;
 }
 
+export function setExternalProvider(provider: ExternalProviderType) {
+  externalProvider = provider;
+}
+
+export function loginWithExternalProvider() {
+  if (externalProvider != null) {
+    accountProvider = externalProvider.getProvider();
```
Comment (arhtudormorar) [inline src/providers/accountProvider.ts:19]: could be only provider, without object calling a member function

File: src/redux/store.ts [hunk 2]
```diff
@@ -56,19 +56,14 @@ export const store = configureStore({
...
-          'accountInfoSlice/setAccountNonce',
-          'extraActionsSlice/initializeExtraActions'
+          'accountInfoSlice/setAccountNonce'
         ],
         ignoredPaths: [
           'payload.nonce',
           'account.account.nonce',
           // extra actions
-          'extraActions.log',
           'payload.log',
```
Comment (arhtudormorar) [inline src/redux/store.ts:65]: could also be removed

File: src/utils/account/loginUserProgramatically.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { loginAction } from 'redux/commonActions';
+import { store } from 'redux/store';
+import { LoginMethodsEnum } from 'types';
+
+export function loginUserProgramatically(
+  address: string,
+  loginMethod = LoginMethodsEnum.extra
```
Comment (arhtudormorar) [inline src/utils/account/loginUserProgramatically.ts:7]: could be handled without any argument

File: src/utils/account/loginUserProgramatically.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+) {
+  store.dispatch(loginAction({ address, loginMethod }));
```
Comment (arhtudormorar) [inline src/utils/account/loginUserProgramatically.ts:9]: { address, loginMethod: LoginMethodsEnum.extra }

File: src/utils/account/loginUserProgramatically.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { loginAction } from 'redux/commonActions';
+import { store } from 'redux/store';
+import { LoginMethodsEnum } from 'types';
+
+export function loginUserProgramatically(
```
Comment (arhtudormorar) [inline src/utils/account/loginUserProgramatically.ts:5]: maybe change name to _loginWithExtraProvider_, to reflect what function actually does

**Review by arhtudormorar — CHANGES_REQUESTED** [review]

**Review by arhtudormorar — APPROVED** [review]

**Pull request merged**